Index: codepress.js
===================================================================
--- codepress.js	(revision 183)
+++ codepress.js	(working copy)
@@ -55,8 +55,8 @@
 		return self.textarea.disabled ? self.editor.getCode() : self.textarea.value;
 	}
 
-	self.setCode = function(code) {
-		self.textarea.disabled ? self.editor.setCode(code) : self.textarea.value = code;
+	self.setCode = function(code,lang) {
+		self.textarea.disabled ? self.editor.setCode(code,lang) : self.textarea.value = code;
 	}
 
 	self.toggleAutoComplete = function() {
Index: engines/gecko.js
===================================================================
--- engines/gecko.js	(revision 183)
+++ engines/gecko.js	(working copy)
@@ -199,11 +199,17 @@
 	// put code inside editor
 	setCode : function() {
 		var code = arguments[0];
+		var lang = arguments[1];
+		if (typeof(lang) != 'undefined'){
+			document.getElementById('cp-lang-style').href = 'languages/' + lang + '.css';
+			document.getElementById('cp-lang-script').src = 'languages/' + lang + '.js';
+		}
 		code = code.replace(/\u2009/gi,'');
 		code = code.replace(/&/gi,'&amp;');
        	code = code.replace(/</g,'&lt;');
         code = code.replace(/>/g,'&gt;');
 		editor.innerHTML = code;
+		CodePress.syntaxHighlight('init');
 	},
 
 	// undo and redo methods
@@ -237,4 +243,4 @@
 }
 
 Language={};
-window.addEventListener('load', function() { CodePress.initialize('new'); }, true);
\ No newline at end of file
+window.addEventListener('load', function() { CodePress.initialize('new'); }, true);
Index: engines/msie.js
===================================================================
--- engines/msie.js	(revision 183)
+++ engines/msie.js	(working copy)
@@ -221,11 +221,18 @@
 	// put code inside editor
 	setCode : function() {
 		var code = arguments[0];
+		var lang = arguments[1];
+		if (typeof(lang) != 'undefined'){
+			document.getElementById('cp-lang-style').href = 'languages/' + lang + '.css';
+			document.getElementById('cp-lang-script').src = 'languages/' + lang + '.js';
+		}
 		code = code.replace(/\u2009/gi,'');
 		code = code.replace(/&/gi,'&amp;');		
        	code = code.replace(/</g,'&lt;');
         code = code.replace(/>/g,'&gt;');
 		editor.innerHTML = '<pre>'+code+'</pre>';
+		
+		CodePress.syntaxHighlight('init');
 	},
 
 	
@@ -260,4 +267,4 @@
 }
 
 Language={};
-window.attachEvent('onload', function() { CodePress.initialize('new');});
\ No newline at end of file
+window.attachEvent('onload', function() { CodePress.initialize('new');});
Index: engines/opera.js
===================================================================
--- engines/opera.js	(revision 183)
+++ engines/opera.js	(working copy)
@@ -218,11 +218,17 @@
 	// put code inside editor
 	setCode : function() {
 		var code = arguments[0];
+		var lang = arguments[1];
+		if (typeof(lang) != 'undefined'){
+			document.getElementById('cp-lang-style').href = 'languages/' + lang + '.css';
+			document.getElementById('cp-lang-script').src = 'languages/' + lang + '.js';
+		}
 		code = code.replace(/\u2009/gi,'');
 		code = code.replace(/&/gi,'&amp;');
        	code = code.replace(/</g,'&lt;');
         code = code.replace(/>/g,'&gt;');
 		editor.innerHTML = code;
+		CodePress.syntaxHighlight('init');
 	},
 
 	// undo and redo methods
